conans/paths.py

```diff
@@ -69,10 +69,18 @@ def conan_expand_user(path):
     return os.path.expanduser(path)
 
 
+def get_conan_user_home():
+    tmp = conan_expand_user(os.getenv("CONAN_USER_HOME", "~"))
+    if not os.path.isabs(tmp):
```

Comment:
Is it possible that ``~`` will give a relative path when expanded?
I don't get the conversion to ``abspath()``, if it is not abs, it will raise.